src/AutoMapper/Mapper.cs

```diff
@@ -28,15 +28,15 @@ public interface IMapperBase
     /// <param name="source">Source object to map from</param>
     /// <param name="destination">Destination object to map into</param>
     /// <returns>The mapped destination object, same instance as the <paramref name="destination"/> object</returns>
-    TDestination Map<TSource, TDestination>(TSource source, TDestination destination);
+    TDestination? Map<TSource, TDestination>(TSource? source, TDestination? destination);
     /// <summary>
     /// Execute a mapping from the source object to a new destination object with explicit <see cref="System.Type"/> objects
     /// </summary>
     /// <param name="source">Source object to map from</param>
     /// <param name="sourceType">Source type to use</param>
     /// <param name="destinationType">Destination type to create</param>
     /// <returns>Mapped destination object</returns>
-    object Map(object source, Type sourceType, Type destinationType);
+    object? Map(object? source, Type? sourceType, Type destinationType);
```

Comment:
Technically `destinationType` can be null here, it would fallback to object. But null doesn't really makes sense, so I want to say this overload is different.